controllers/operandrequest/operandrequest_controller.go

```diff
@@ -84,6 +84,28 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reconcileErr er
 		}
 	}()
 
+	// Remove finalizer when DeletionTimestamp none zero
+	if !requestInstance.ObjectMeta.DeletionTimestamp.IsZero() {
+
+		// Check and clean up the subscriptions
+		err := r.checkFinalizer(ctx, requestInstance)
+		if err != nil {
+			klog.Errorf("failed to clean up the subscriptions for OperandRequest %s: %v", req.NamespacedName.String(), err)
+			return ctrl.Result{}, err
```

Comment:
@horis233 Would you mind explaining it a little bit more? Why we move this section above the `add finalizer` would help with it?

From my understanding, if we directly delete the namespace, there are some other Common Service resources with the finalizer pending in the namespace, causing the situation that we could not get `err==nil` from `checkFinalizer` which means some resources cannot be deleted.

Reply:
We move this logic before the permission check is because when the namespace is deleted, ODLM will lose the permission from the namespace scope operator. It can't pass the permission check. However, if there is nothing to delete or operand and operand are not in the deleting namespace, ODLM should be able to remove the finalizer from the operandrequest. That is the reason why we need this PR.

Reply:
Oh I see, usually the cloud pak create OperandRequest in the their own namespace, and the operators and operands are installed in `ibm-common-services` namespace, In this case, it could be correctly handled.

Thank you!

Reply:
Yes, exactly